test/job.js

```diff
@@ -212,6 +226,78 @@ describe('Job', () => {
       expect(moment(job.attrs.nextRunAt).toDate().getDate()).to.be(moment(job.attrs.lastRunAt).add(1, 'days').toDate().getDate());
     });
 
+    it('set after startDate when startDate is in future ', () => {
+      const now = new Date('2015-01-01T06:00:00-00:00');
+      const _start = new Date('2015-01-03');
+      job.attrs.lastRunAt = now;
+      job.repeatEvery('0 6 * * *', {
+        timezone: 'GMT',
+        startDate: _start
+      });
+      job.computeNextRunAt();
+      expect(job.attrs.nextRunAt.valueOf()).to.greaterThan(_start.valueOf());
+    });
+
+    it('set to next occurence when startDate is in past ', () => {
+      const now = new Date();
+      const _start = new Date('2018-04-17T18:00:00-00:00');
+      job.attrs.lastRunAt = null;
+      job.repeatEvery('0 6 * * *', {
+        timezone: 'GMT',
+        startDate: _start
+      });
+      job.computeNextRunAt();
+      expect(job.attrs.nextRunAt.valueOf()).to.greaterThan(now.valueOf());
+    });
+
+    it('set to next occurence when startDate is in past and interval is humanInterval', () => {
+      const now = new Date();
+      const _start = new Date('2018-04-17T06:00:00-00:00');
+      job.attrs.lastRunAt = null;
+      job.repeatEvery('day', {
+        timezone: 'GMT',
+        startDate: _start
+      });
+      job.computeNextRunAt();
+      expect(job.attrs.nextRunAt.valueOf()).to.greaterThan(now.valueOf());
+    });
+
+    it('set at startDate when startDate is in future and interval is humanInterval ', () => {
+      const now = new Date('2015-01-01T06:00:00-00:00');
+      const _start = new Date('2015-01-03');
+      job.attrs.lastRunAt = now;
+      job.repeatEvery('5 minutes', {
+        timezone: 'GMT',
+        startDate: _start
+      });
+      job.computeNextRunAt();
+      expect(job.attrs.nextRunAt.valueOf()).to.be(_start.valueOf());
```

Comment:
I have a suspicion that this is hiding an issue where `job.attrs.nextRunAt` is now a number, and not a date.

Could all the date comparisons change to use `expect(dateObject).to.eql(new Date('expected-date-string'))`?